target_snowflake/load_via_snowpipe.py

```diff
@@ -0,0 +1,149 @@
+                query_stage_master_key=master_key,
+                query_id='',
+                smk_id=0
+            )
+            encryption_metadata, encrypted_file = SnowflakeEncryptionUtil.encrypt_file(
+                encryption_material,
+                file,
+                tmp_dir=temp_dir
+            )
+
+            # Upload to s3
+            extra_args = {'ACL': s3_acl} if s3_acl else dict()
+
+            # Send key and iv in the metadata, that will be required to decrypt and upload the encrypted file
+            extra_args['Metadata'] = {
+                'x-amz-key': encryption_metadata.key,
+                'x-amz-iv': encryption_metadata.iv
+            }
+            self.s3.upload_file(encrypted_file, bucket, s3_key, ExtraArgs=extra_args)
+
+            # Remove the uploaded encrypted file
+            os.remove(encrypted_file)
+
+        # Upload to S3 without encrypting
+        else:
+            extra_args = {'ACL': s3_acl} if s3_acl else None
+            self.s3.upload_file(file, bucket, s3_key, ExtraArgs=extra_args)
+
+        return s3_key,s3_folder_name
+
+    def load_via_snowpipe(self,s3_key,s3_folder_name):
+        obj_name = s3_folder_name.replace('/','')
+        pipe_name = "{0}.{1}.{2}_s3pipe".format(self.connection_config['dbname'],self.schema_name,obj_name)
+
+        create_pipe_sql = """create pipe if not exists {0} as
+                            copy into {1}.{2}.{3} from @{1}.{4}
+                            file_format = (format_name =  {1}.{5} );""".format(
+                                pipe_name,
+                                self.connection_config['dbname'],
+                                self.schema_name,
+                                obj_name,
+                                self.connection_config['stage'],
+                                self.connection_config['file_format'])
+
+        pipe_status_sql = "select system$pipe_status('{}');".format(pipe_name)
+
+        with self.open_connection() as connection:
+            with connection.cursor(snowflake.connector.DictCursor) as cur:
+                # Create snowpipe if it does not exist
+                try:
+                    cur.execute(pipe_status_sql)
+                    self.logger.info("""snowpipe "{}" already exists!!""".format(pipe_name))
+                except snowflake.connector.errors.ProgrammingError as excp:
+                    self.logger.info("""snowpipe "{}" does not exist. Creating...""".format(pipe_name))
+                    cur.execute(create_pipe_sql)
+                finally:
+                    cur.close()
+        
+        logging.basicConfig(filename='/tmp/ingest.log',level=logging.DEBUG)
+        
+        logger = getLogger(__name__)
+
+        # If you generated an encrypted private key, implement this method to return
+        # the passphrase for decrypting your private key.
+        def get_private_key_passphrase():
+            return "CkYt-3fngCdm65P2"
```

Comment:
this should use an environment variable to populate and handle the case when passphrase is empty.